### PR TITLE
Update the data server to use esm modules for everything

### PR DESCRIPTION
**Pull request by zusorio**
I've tested this and everything seems to work. This is a first step for potentially introducing some typescript stuff later on

### CODE REVIEW

**Comment by slmnio** [comment]
Very very nice

**Comment by zusorio** [comment]
Just noticed that discord/staff.js still needs to be updated

File: server/src/action-utils/action-manager.js [hunk 1]
```diff
@@ -24,16 +19,16 @@ async function loadActions(directory) {
...
 
 let managers = {};
 
-async function load(expressApp, cors, Cache, io) {
+export async function load(expressApp, cors, Cache, io) {
     const actionApp = express.Router();
     actionApp.use(bodyParser.json());
     actionApp.options("/*", cors());
-    actions = (await loadActions(path.join(__dirname, "..", "actions"))) || [];
+    actions = (await loadActions(path.join(import.meta.dirname, "..", "actions"))) || [];
```
Comment (slmnio) [inline server/src/action-utils/action-manager.js:31]: This fails on Node 18

Comment (slmnio) [reply server/src/action-utils/action-manager.js:31]: The build processes on Actions and the server itself is on 18.20.2. We _can_ up it if we have to, but this doesn't look like something that's worth bumping a node version.

Comment (zusorio) [reply server/src/action-utils/action-manager.js:31]: I think there was some less elegant way of doing it before node 20, can look into this (later)

File: server/src/discord/staff.js [hunk 2]
```diff
@@ -1,16 +1,16 @@
 /* eslint-disable */
 /* BPL Staff Automation */
 
-const client = require("./client.js");
+import client from "./client.js";
 if (!client) return console.warn("Staff application system will not be set up because no Discord key is set.");
```
Comment (slmnio) [inline server/src/discord/staff.js:5]: Need to remove this

File: server/src/index.js [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
+import routes from "./routes.js";
+import images from "./images.js";
+import discordAuth from "./discord/auth.js";
+import webAuction from "./web_auction.js";
 
 /* The staff module should only run on the server, probably not your local machine. */
 let staffKeysRequired = ["DISCORD_TOKEN", "STAFFAPPS_GUILD_ID", "STAFFAPPS_CATEGORY_ID", "STAFFAPPS_APPLICATION_CHANNEL_ID", "IS_SLMNGG_MAIN_SERVER"];
 if (staffKeysRequired.every(key => process.env[key])) {
-    require("./discord/staff.js");
+    await import("./discord/staff.js");
```
Comment (slmnio) [inline server/src/index.js:19]: This should be all we need

File: server/src/discord/staff.js (hunk 2 above)
Comment (slmnio) [reply server/src/discord/staff.js:5]: I don't know what I meant by this

**Comment by slmnio** [comment]
Looks all good, I'm working with working Typescript in #275

**Pull request merged**